src/server.ml

```diff
@@ -337,10 +345,24 @@ module Make (IO : IO) = struct
           server_request:_ ->
           id:Req_id.t ->
           r Lsp.Client_request.t ->
-          r IO.t =
+          (r, string) result IO.t =
         fun ~notify_back ~server_request ~id (r : _ Lsp.Client_request.t) ->
           Trace.with_span ~__FILE__ ~__LINE__ "linol.on-request"
-          @@ fun _sp : r IO.t ->
+          @@ fun _sp : (r, string) result IO.t ->
```

Comment:
this is not on you to fix but we need a async `Trace.with_span` for sure :/

Reply:
Yeah, that's a good point. Does something of that sort exist elsewhere or would we have to create that? Would it basically mean putting a lock on the collector?

Reply:
it's a bit annoying because we're functorized over IO, so the functor application would need to provide a tracing helper even for users who don't care. Out of scope for this PR for sure.

Reply:
Oh right, that's very annoying indeed :-/